test/e2e/knative_serving_test.go

```diff
@@ -36,16 +37,25 @@ func TestKnativeServing(t *testing.T) {
 				t.Fatalf("Deployment %s is not ready: %v", deployment, err)
 			}
 		}
-		// Check the desired scale of deployments in the ingress namespace.
+
 		ingressDeployments := []string{"net-kourier-controller", "3scale-kourier-gateway"}
+		ingressNamespace := servingNamespace + "-ingress"
+
+		// If FULL_MESH is true, net-istio is used instead of net-kourier.
+		if os.Getenv("FULL_MESH") == "true" {
+			ingressDeployments = []string{"net-istio-controller", "net-istio-webhook"}
+			ingressNamespace = servingNamespace
+		}
+
+		// Check the desired scale of deployments in the ingress namespace.
 		for _, deployment := range ingressDeployments {
-			if err := test.CheckDeploymentScale(caCtx, servingNamespace+"-ingress", deployment, haReplicas); err != nil {
+			if err := test.CheckDeploymentScale(caCtx, ingressNamespace, deployment, haReplicas); err != nil {
 				t.Fatalf("Failed to verify default HA settings: %v", err)
 			}
 		}
 		// Check the status of deployments in the ingress namespace.
 		for _, deployment := range ingressDeployments {
-			if _, err := test.WithDeploymentReady(caCtx, deployment, servingNamespace+"-ingress"); err != nil {
+			if _, err := test.WithDeploymentReady(caCtx, deployment, ingressNamespace); err != nil {
```

Comment:
Can't we have one loop here?

Reply:
Yes, we can. But it should be aligned with other codes like above L27-39  and the same test in `knative_eventing_test.go`. So if we want to change it, I think another refactoring PR is better.

Reply:
ok no problem let's get this in.

Reply:
The above lines L27-39 have two loops because the sets of pods are different in each loop. I believe for this ingress stuff we can have just one loop because the sets are same. Btw, the sets are also different in Eventing tests. So, this is probably the only place where it can be merged into one loop.

Reply:
But it can be done in a different PR, of course :)

Reply:
Indeed @mgencur, @nak3 if you can fix it now would be nice.

Reply:
> the sets of pods are different in each loop

I think we can use the same set of pods(deployments) now. Upstream operator supports all deployments by HA config. I have opened for it in https://github.com/openshift-knative/serverless-operator/issues/1377

Reply:
ok let's deal with this in another pr.